mailpoet/lib/Automation/Engine/Data/StepRunArgs.php

```diff
@@ -39,6 +46,11 @@ public function __construct(
       $key = $subject->getKey();
       $this->subjectEntries[$key] = array_merge($this->subjectEntries[$key] ?? [], [$entry]);
       $this->subjectKeyClassMap[get_class($subject)] = $key;
+
+      foreach ($subject->getFields() as $field) {
+        $this->fields[$field->getKey()] = $field;
+        $this->fieldToSubjectMap[$field->getKey()] = $key;
```

Comment:
> also, what i am just thinking: currently our code base can't provide for that (and maybe it shouldnt) but how would we handle SubscriberSubject[] where one subject might match and the other does not?

We don't allow multiple subjects of the same type at all currently. I feel like enabling that would be quite a challenge on multiple levels. The automation format doesn't limit that (subjects field is an array, not key-value), but I'm not sure if we should ever allow that. Maybe it will be better to define some "multi-subjects" instead. Something like `SubscriberGroupSubject` or a more generic `SubjectGroup`.